client/components/Editor/index.jsx

```diff
@@ -69,6 +89,43 @@ class Editor extends Component {
         this.props.history.push('/');
     }
 
+    async updateScenario(updates = {}) {
+        if (this.state.saving) {
+            return;
+        }
+
+        this.setState({ saving: true });
+
+        // NOTE: this is to support saving the whole
+        //       scenario when clicking the [save icon]
+        //       that's displayed via EditorMenu.
+        const data = {
+            title: this.props.title,
+            description: this.props.description,
+            categories: this.props.categories,
+            status: this.props.status
+        };
+
+        Object.assign(data, updates);
+
+        const submitCallback = this.getSubmitCallback();
+        const response = await (await submitCallback(data)).json();
+
+        this.setState({ saving: false });
+
+        switch (response.status) {
+            case 200:
```

Comment:
Will we ever need to support a 201 status too? I think saving a scenario for the first time returns a 201 status.

Reply:
This path shouldn't be taken for the initial save 🤷‍♀